include/smack/Prelude.h

```diff
@@ -55,41 +61,48 @@ struct FpOp {
   static const Attr* fpAttrFunc(std::string opName);
 };
 
+// builtin functions templated by a function type. A function
+// of such type, when applied, returns an attribute
 template<typename ATTRT>
 struct BuiltinOp : public Op {
   typedef const Attr* (*attrT)();
   attrT func;
-  BuiltinOp(ATTRT func) : Op(B), func((attrT)func) {}
+  BuiltinOp(ATTRT func) : Op(Builtin), func((attrT)func) {}
   static bool classof(const Op* op) {
-    return op->getOpType() == B;
+    return op->getOpType() == Builtin;
   }
 };
 
+// inlined functions templated by a function type. A function
+// of such type, when applied, returns an expression as the function body
 template<typename EXPRT>
 struct InlinedOp : public Op {
   typedef const Expr* (*exprT)();
   exprT func;
-  InlinedOp(EXPRT func) : Op(I), func((exprT)func) {}
+  InlinedOp(EXPRT func) : Op(Inlined), func((exprT)func) {}
   static bool classof(const Op* op) {
-    return op->getOpType() == I;
+    return op->getOpType() == Inlined;
   }
 };
 
+// uninterpreted functions
 struct UninterpretedOp : public Op {
-  UninterpretedOp() : Op(U) {}
+  UninterpretedOp() : Op(Uninterpreted) {}
   static bool classof(const Op* op) {
-    return op->getOpType() == U;
+    return op->getOpType() == Uninterpreted;
   }
 };
 
 class Prelude;
+// generic generater class that produces relevant declarations
```

Comment:
This should be `generator`. Fix everywhere.

Reply:
Done.